contracts/cctp/libs/Request.sol

```diff
@@ -0,0 +1,195 @@
+/// | ---------- | --------------- | ------- | ----- | -------------------------------------------------- |
+/// | [084..104) | pool            | address | 20    | Liquidity pool for swapping Circle token           |
+/// | [104..105) | tokenIndexFrom  | uint8   | 1     | Index of the minted Circle token in the pool       |
+/// | [105..106) | tokenIndexTo    | uint8   | 1     | Index of the final token in the pool               |
+/// | [106..116) | deadline        | uint80  | 10    | Latest timestamp to execute the swap               |
+/// | [116..148) | minAmountOut    | uint256 | 32    | Minimum amount of tokens to receive from the swap  |
+library RequestLib {
+    uint32 internal constant REQUEST_BASE = 0;
+    uint32 internal constant REQUEST_SWAP = 1;
+
+    uint256 private constant OFFSET_ORIGIN_DATA = 0;
+    uint256 private constant OFFSET_AMOUNT = OFFSET_ORIGIN_DATA + 32;
+    uint256 private constant OFFSET_RECIPIENT = OFFSET_AMOUNT + 32;
+    uint256 private constant REQUEST_BASE_LENGTH = OFFSET_RECIPIENT + 20;
+
+    uint256 private constant OFFSET_SWAP_PARAMS = REQUEST_BASE_LENGTH;
+    uint256 private constant OFFSET_MIN_AMOUNT_OUT = OFFSET_SWAP_PARAMS + 32;
+    uint256 private constant REQUEST_SWAP_LENGTH = OFFSET_MIN_AMOUNT_OUT + 32;
+    uint256 private constant SWAP_PARAMS_LENGTH = REQUEST_SWAP_LENGTH - REQUEST_BASE_LENGTH;
+
+    /// @notice Formats the base request into a bytes array.
+    /// @param originDomain_        Domain of the origin chain
+    /// @param nonce_               Nonce of the CCTP message on origin domain
+    /// @param originBurnToken_     Circle token that was burned on origin domain
+    /// @param amount_              Amount of tokens to burn
+    /// @param recipient_           Recipient of the tokens on destination domain
+    /// @return formattedRequest    Properly formatted base request
+    function formatBaseRequest(
+        uint32 originDomain_,
+        uint64 nonce_,
+        address originBurnToken_,
+        uint256 amount_,
+        address recipient_
+    ) internal pure returns (bytes memory formattedRequest) {
+        formattedRequest = abi.encodePacked(originDomain_, nonce_, originBurnToken_, amount_, recipient_);
+    }
+
+    /// @notice Formats the swap parameters part of the swap request into a bytes array.
+    /// @param pool_                Liquidity pool for swapping Circle token
+    /// @param tokenIndexFrom_      Index of the minted Circle token in the pool
+    /// @param tokenIndexTo_        Index of the final token in the pool
+    /// @param deadline_            Latest timestamp to execute the swap
+    /// @param minAmountOut_        Minimum amount of tokens to receive from the swap
+    /// @return formattedSwapParams Properly formatted swap parameters
+    function formatSwapParams(
+        address pool_,
+        uint8 tokenIndexFrom_,
+        uint8 tokenIndexTo_,
```

Comment:
`amountIn_` parameter necessary here? From peeking at the draft PR #234 seems like you'll potentially infer the amount in from the mint, but checking to be sure.

Reply:
Not necessary indeed! This will be calculated on destination chain as `amount` (present in Base Request) minus the bridge fees.